common/datadogSharedLogic.ts

```diff
@@ -25,7 +25,8 @@ export function validateProps(props: DatadogProps, apiKeyArnOverride = false) {
   if (
     props.site !== undefined &&
     !siteList.includes(props.site.toLowerCase()) &&
-    !(props.site.startsWith("${Token[") && props.site.endsWith("]}"))
+    !(props.site.startsWith("${Token[") && props.site.endsWith("]}")) &&
+    process.env.DD_CDK_BYPASS_SITE_VALIDATION !== "true"
```

Comment:
So this will still throw an error if the user specifies `DD_CDK_BYPASS_SITE_VALIDATION=True` or `DD_CDK_BYPASS_SITE_VALIDATION=TRUE`.

I think we want to lowercase that?

Worth considering this solution versus checking for this to be `false`, or even checking for the presence of the env var at all?

(to be clear I think we should lowercase and call it good, I'm just laying out the alternative as a thought exercise)

Reply:
Good catch! I will lowercase the env variable first.

Note that if the environment variable is not set at all, set to `false`, or set to anything but `true`, the site check will still throw an error.
If we instead checked `process.env. DD_CDK_BYPASS_SITE_VALIDATION === "false"`, then the site check would be bypassed unless they set DD_CDK_BYPASS_SITE_VALIDATION to false.

Reply:
Exactly - it's just something to think about. Would anyone actually set `DD_CDK_BYPASS_SITE_VALIDATION` to false? Probably not. They'd only set the env var if they actually want to bypass the site check (eg: the intent of the config var is declared by the presence of the env var).

We see this in other cases too like `DD_TRACING_ENABLED`. But compare that to an env var that is not a boolean, like `DD_SITE` where we have different assumptions. We may consider checking just for `presence` here versus `value`.

Reply:
Ah, I see your point. If we want to check for only presence, I can do something like:
`&& !process.env.DD_CDK_BYPASS_SITE_VALIDATION`.

Which option do you think would be better? I'm open to doing both

Reply:
Ohhh I see. In this specific case I think let's be consistent with `datadog-ci` then.